vmb/matrimony/models/relations.py

```diff
@@ -174,3 +178,25 @@ class Meta:
 
     def __str__(self):
         return f"{self.name}"
+
+
+class CurrencyExchangeRate(BaseModel):
+    from_currency = models.CharField(max_length=3, null=True)
+    to_currency = models.CharField(max_length=3, null=True, default="INR")
+    exchange_rate = models.DecimalField(max_digits=10, decimal_places=2, null=True)
+
+    def get_exchange_rate(self):
+        if self.exchange_rate == None:
+            backend = OpenExchangeRatesBackend(url=OPEN_EXCHANGE_RATES_URL)
+            backend.update_rates(
+                symbols=",".join([self.from_currency, self.to_currency])
+            )
+            self.exchange_rate = get_rate(
+                self.from_currency,
+                self.to_currency,
+                backend=OpenExchangeRatesBackend.name,
+            )
+        return self.exchange_rate
```

Comment:
How's the fetched `exchange_rate` getting saved?

Reply:
in the line self.exchange_rate= get_rate(
                self.from_currency,
                self.to_currency,
                backend=OpenExchangeRatesBackend.name,
            )

Reply:
however this should have a self.save() add the end to save the changes in the object. Missed it. Thank You for highlighting that.